src/python/pants/base/exception_sink.py

```diff
@@ -369,8 +384,15 @@ def log_exception(cls, exc_class=None, exc=None, tb=None, add_newline=False):
             extra_err_msg = "Additional error logging unhandled exception {}: {}".format(exc, e)
             logger.error(extra_err_msg)
 
-        # Generate an unhandled exception report fit to be printed to the terminal.
-        logger.exception(exc)
+        # The rust logger implementation is used for most of pants's execution, but at import time,
+        # we want to be able to see any stacktrace to know where the error is being raised.
+        if cls._logging_initialized:
+            logger.exception(exc)
+        else:
+            exception_log_entry = cls._format_unhandled_exception_log(
```

Comment:
What happens if we get rid of the logic for checking and setting `cls._logging_initialized` and just *always* use the `cls._format_unhandled_exception_log` form? If I understand https://docs.python.org/3.6/library/logging.html#logging.Logger.exception correcty, `logger.exception` and `logger.error` should both be writing fairly similar text to the Rust logging faculty, once it's been initialized. If we could get away with just modifying this one line to print the additional context we need for import-time errors, without adding any additional logic to think about in this class, I think that would be ideal.

Reply:
That works, but why doesn't it print the stacktrace twice when we send the rust logger the stacktrace? What is it doing such that sending it the exception vs a completely different string with the exception + stacktrace produces the same output?

Reply:
> That works, but why doesn't it print the stacktrace twice when we send the rust logger the stacktrace? What is it doing such that sending it the exception vs a completely different string with the exception + stacktrace produces the same output?

I'm not sure. Does anything change if you call `logger.error` with the string generated by `format_unhandled_exception_log` rather than `logger.exception`?

Reply:
No! Which is so weird! I would like to understand this part further, because it's not immediately clear to me, but I wasn't able to easily figure it out by scanning the rust logger implementation, and the right behavior still seems to occur on `--no-print-exception-stacktrace`.